ExperimentTwitter/TweetsTableViewController.m

```diff
@@ -0,0 +1,174 @@
+//
+//  TweetsViewController.m
+//  ExperimentTwitter
+//
+//  Created by ayur.j on 25/08/17.
+//  Copyright © 2017 ayur.j. All rights reserved.
+//
+
+#import "TweetsTableViewController.h"
+#import "PostTweetViewController.h"
+#import "CustomTweetCell.h"
+#import "TwitterClient.h"
+#import "Tweet+Twitter.h"
+#import "TweetsSearchViewController.h"
+
+@interface TweetsTableViewController ()
+@property (strong, nonatomic) NSMutableArray *tweets;
+@property (strong, nonatomic) UIRefreshControl *refreshControl;
+@property (nonatomic) BOOL loadMoreData;
+@end
+
+@implementation TweetsTableViewController
+
+- (id) initWithTweets:(NSArray *)tweets {
+    self = [super init];
+    self.tweets = [NSMutableArray arrayWithArray:tweets];
+    return self;
+}
+
+@synthesize refreshControl;
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
+{
+    return [self.tweets count];
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView
+         cellForRowAtIndexPath:(NSIndexPath *)indexPath;
+{
+    CustomTweetCell *cell = [tableView dequeueReusableCellWithIdentifier:@"tweetCell"];
+    Tweet *tweet = self.tweets[indexPath.row];
+    cell.tweet = tweet;
+    cell.replyButton.tag = indexPath.row;
+    [cell.replyButton addTarget:self action:@selector(onReplyButtonPress:) forControlEvents:UIControlEventTouchUpInside];
+    cell.selectionStyle = UITableViewCellSelectionStyleNone;
+    return cell;
+}
+
+
+- (void)tableView:(UITableView *)tableView
+  willDisplayCell:(UITableViewCell *)cell
+forRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    if(indexPath.row == (self.tweets.count - 5) && self.loadMoreData) {
+        [[TwitterClient sharedInstance]tweetsWithParams:@{@"endPoint":[self getEndPointWithMaxIdParameter:[self getMaxIdParameterFromLastTweet]]}
+                                             completion:^(NSArray *tweets, NSError *error) {
+            if(error == nil) {
+                if([tweets count] > 0) {
+                    [self sortTweetsListByCreatedAt:&tweets];
+                    [self.tweets addObjectsFromArray:tweets];
+                    [self.tableView reloadData];
+                }
+                else {
+                    self.loadMoreData = NO;
+                }
+            }
+        }];
+    }
+}
+
+- (void) fetchMoreTweets {
+    [[TwitterClient sharedInstance]tweetsWithParams:@{@"endPoint":[self getEndPointWithMaxIdParameter:[self getMaxIdParameterFromLastTweet]]}
+                                         completion:^(NSArray *tweets, NSError *error) {
+        if(error == nil) {
+            if([tweets count] > 0) {
+                [self sortTweetsListByCreatedAt:&tweets];
+                [self.tweets addObjectsFromArray:tweets];
+                [self.tableView reloadData];
+            }
+            else {
+                self.loadMoreData = NO;
```

Comment:
Code in `fetchMoreTweets` and `...willDisplayCell...` is duplicated. Can be extracted out to a method.